addons/im_livechat/static/src/embed/common/thread_model_patch.js

```diff
@@ -23,6 +23,19 @@ patch(Thread, threadStaticPatch);
 patch(Thread.prototype, {
     setup() {
         super.setup();
+        this.livechat_operator_id = Record.one("Persona", {
+            onUpdate() {
+                if (!this.livechat_operator_id) {
+                    return;
+                }
+                const ONE_DAY_TTL = 60 * 60 * 24;
+                expirableStorage.setItem(
+                    "im_livechat_previous_operator",
+                    this.livechat_operator_id.id,
+                    ONE_DAY_TTL * 7
+                );
+            },
+        });
```

Comment:
That's way too much of a shortcut.
There is no guaranteed any given thread is actually the last relevant one.
Especially if operator sees this field themselves.
This should be done in flows instead, or with a dedicated field "set last operator" which is only set when necessary.